### PR TITLE
Capture HTTP Request Method

### LINKED ISSUE #611: Capture HTTP request method on transactions
It is valuable for customers to be able to distinguish between the different type of types of HTTP request methods. We will implement support for this by capturing the request method as defined in the agent [spec](https://source.datanerd.us/agents/agent-specs/blob/master/APIs/web_transaction.md).

### PR DESCRIPTION
**Pull request by tehbio**
### Description
  
Resolves #611 by adding a new `request.method` attribute to the Agent, and implementing its capture in the `AspNetCore`, `Asp35`, `Wcf3`, and `Owin` wrappers.

### Testing

Leveraged the new request capturing integration tests, adding a simple method assertion to each.

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Core/Transactions/TransactionMetadata.cs [hunk 1]
```diff
@@ -101,6 +104,11 @@ public IImmutableTransactionMetadata ConvertToImmutableMetadata()
         public bool IsSynthetics => !string.IsNullOrEmpty(_syntheticsResourceId) && !string.IsNullOrEmpty(_syntheticsJobId) &&
                                      !string.IsNullOrEmpty(_syntheticsMonitorId);
 
+        public void SetRequestMethod(string requestMehtod)
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Transactions/TransactionMetadata.cs:107]: requestMehtod => requestMethod

File: src/Agent/NewRelic/Agent/Core/Transactions/NoOpTransaction.cs [hunk 2]
```diff
@@ -179,6 +179,8 @@ public void SetCustomTransactionName(string name, TransactionNamePriority priori
 
         }
 
+        public void SetRequestMethod(string requestMethod) { }
+
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Transactions/NoOpTransaction.cs:183]: suggest using consistent formatting within a file... all the other methods could be converted to this concise format IMHO

File: src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs [hunk 3]
```diff
@@ -775,6 +775,14 @@ public void RollupTransactionNameByStatusCodeIfNeeded()
             }
         }
 
+        public void SetRequestMethod(string requestMethod)
+        {
+            if (requestMethod == null)
+                throw new ArgumentNullException(nameof(requestMethod));
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs:781]: Not sure what are coding standards say, but I would add braces to all logic flow control

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ITransaction.cs [hunk 4]
```diff
@@ -216,6 +216,13 @@ public interface ITransaction
         /// <exception cref="ArgumentNullException"></exception>
         void SetCustomTransactionName(string name, TransactionNamePriority priority = TransactionNamePriority.Uri);
 
+        /// <summary>
+        /// Set the Request Method for the current transaction (if there is one).
+        /// </summary>
+        /// <param name="requestMethod">The Request Method for this transaction. Must not be null.</param>
+        /// <exception cref="ArgumentNullException"></exception>
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ITransaction.cs:223]: are we sure it throws? I don't see parameter validation anywhere

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
The only thing I really care about changing is fixing teh typo. Up to you what you do with the rest of the feedback. I like the test coverage!

File: src/Agent/NewRelic/Agent/Core/Transactions/TransactionMetadata.cs (hunk 1 above)
Comment (tehbio) [reply src/Agent/NewRelic/Agent/Core/Transactions/TransactionMetadata.cs]: Nice catch!

**Review by JcolemanNR — APPROVED** [review]
Ship it (assuming the tests pass)

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ITransaction.cs (hunk 4 above)
Comment (tehbio) [reply src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Api/ITransaction.cs:223]: Happens in the implementation here:
https://github.com/newrelic/newrelic-dotnet-agent/blob/ec3a5800a2d9844f07243e8c919294d059c2b901/src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs#L778-L781

File: src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs (hunk 3 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs:781]: Our coding standard do prefer braces over the shorthand.

**Review by jaffinito — COMMENTED** [review]
Other than the code standards comment, this looks good

**Comment by JcolemanNR** [comment]
I ❤️ that the bounded integration tests all passed.

**Pull request merged**